modules/openy_features/openy_node/modules/openy_node_landing/src/LandingPageAlertBuilder.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\openy_node_landing;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\openy_node_alert\Service\AlertBuilderInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Provides an alert builder for landing pages.
+ */
+class LandingPageAlertBuilder implements AlertBuilderInterface {
+
+  /**
+   * The node storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $nodeStorage;
+
+  /**
+   * Constructs the LandingPageAlertBuilder.
+   *
+   * @param EntityTypeManagerInterface $entity_type_manager
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->nodeStorage = $entity_type_manager->getStorage('node');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(EntityInterface $node) {
+    if ($node->bundle() != 'landing_page') {
+      return;
+    }
+    if ($node->field_lp_location->isEmpty()) {
+      return;
+    }
+    $location_ids = array_column($node->field_lp_location->getValue(), 'target_id');
+    if (empty($location_ids)) {
+      return;
+    }
```

Comment:
Love the early return here; still it should be moved to a separate applies() method. And build should be just building - assuming the applies() was already checked and returned true.

Reply:
Moved to apply() function